server/src/main/java/org/elasticsearch/bootstrap/JNANatives.java

```diff
@@ -260,4 +269,39 @@ static void tryInstallSystemCallFilter(Path tmpFile) {
             logger.warn("unable to install syscall filter: ", e);
         }
     }
+
+    @SuppressForbidden(reason = "need access to fd on FileOutputStream")
+    static void fallocateSnapshotCacheFile(Environment environment, long fileSize) throws IOException {
```

Comment:
It's a little unfortunate to leak that much logic into this class, but I figured abstracting things away further to isolate the `fallocate` as much as possible wasn't really worth the extra complication (IMO).